keyboards/edinburgh41/thumbstick.c

```diff
@@ -0,0 +1,114 @@
+/* Copyright 2022 schwarzer-geiger
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation, either version 2 of the License, or
+ * (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ */
+
+// import thumbstick settings from config
+#pragma once
+
+#include "config.h"
+
+bool cursor_mode = false;
+bool scrolling_mode = false;
+bool tapping_mode = false;
+
+// tracks if thumbstick was released
+bool returned_to_zero = true;
+
+// tracks how many times mouse_report.x/y have been read zero in succession
+int zero_reads = 0;
+
+// manipulate mouse report based on current mode
+report_mouse_t pointing_device_task_user(report_mouse_t mouse_report) {
```

Comment:
These should not be using user functions.  It should be using kb functions, with the user call, or this should be moved entirely to the keymap.c files